sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -1317,5 +1318,47 @@ async def read_feed_ranges(
                 [Range("", "FF", True, False)],
                 **kwargs)
 
-        return [FeedRangeEpk(Range.PartitionKeyRangeToRange(partitionKeyRange))
+        return [FeedRangeEpk(Range.PartitionKeyRangeToRange(partitionKeyRange), self.container_link)
                 for partitionKeyRange in partition_key_ranges]
+
+    def get_updated_session_token(self,
+                                  feed_ranges_to_session_tokens: List[Tuple[FeedRange, str]],
```

Comment:
PR description has sample of how it could be used. There are two main scenarios storing by feed range representing logical partition or by feed range representing physical partition. If they want to store their session tokens by logical partition, they would use the `feed_range_from_partition_key` api I exposed and get the session token from the response and create this list of tuples. The other option is them storing the session tokens by physical partition using the `read_feed_ranges` api to get the feed ranges than they have to use the `is_subset` api to figure out which partition key is in each range and then combine that with the session token from the response.